vowpalwabbit/expreplay.h

```diff
@@ -92,8 +92,8 @@ LEARNER::base_learner* expreplay_setup(arguments& arg)
   if (! arg.all->quiet)
     std::cerr << "experience replay level=" << er_level << ", buffer=" << er->N << ", replay count=" << er->replay_count << std::endl;
 
-  er->base = setup_base(arg);
-  LEARNER::learner<expreplay>* l = &init_learner(er, er->base, predict_or_learn<true,lp>, predict_or_learn<false,lp>);
+  er->base = LEARNER::as_singleline(setup_base(arg));
```

Comment:
Maybe just overload setup_base instead?

Reply:
Todo: typecheck the base type against the single/multiline type expected by predict_or_learn.

Reply:
1) Added typecheck for base type against single/multiline type expected by predict_or_learn.
2) Cannot overload setup_base because it only differs by return type.